engine/transaction/record/record_outline.go

```diff
@@ -0,0 +1,126 @@
+package record
+
+import (
+	"context"
+
+	"github.com/bitcoin-sv/go-sdk/spv"
+	trx "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/conv"
+	"github.com/bitcoin-sv/spv-wallet/engine/database"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/engine/transaction"
+	txerrors "github.com/bitcoin-sv/spv-wallet/engine/transaction/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/transaction/outlines"
+	"github.com/bitcoin-sv/spv-wallet/models/bsv"
+	"github.com/bitcoin-sv/spv-wallet/models/transaction/bucket"
+)
+
+// RecordTransactionOutline will validate, broadcast and save a transaction outline
+func (s *Service) RecordTransactionOutline(ctx context.Context, outline *outlines.Transaction) error {
+	tx, err := trx.NewTransactionFromBEEFHex(outline.BEEF)
+	if err != nil {
+		return txerrors.ErrTxValidation.Wrap(err)
+	}
+
+	if ok, err := spv.VerifyScripts(tx); err != nil {
+		return txerrors.ErrTxValidation.Wrap(err)
+	} else if !ok {
+		return txerrors.ErrTxValidation
+	}
+
+	utxos, err := s.getTrackedUTXOsFromInputs(ctx, tx)
+	if err != nil {
+		return err
+	}
+
+	txID := tx.TxID().String()
+	for _, utxo := range utxos {
+		utxo.Spend(txID)
+	}
+
+	var newOutputs []*database.Output
+	var newDataRecords []*database.Data
+	if outline.Annotations != nil {
+		newOutputs, newDataRecords, err = s.processAnnotatedOutputs(tx, *outline.Annotations)
+		if err != nil {
+			return err
+		}
+	}
+
+	if err = s.broadcaster.Broadcast(ctx, tx); err != nil {
+		return txerrors.ErrTxBroadcast.Wrap(err)
+	}
+
+	txRecord := database.Transaction{
+		ID:       txID,
+		TxStatus: database.TxStatusBroadcasted,
+	}
+
+	upsertOutputs := append(newOutputs, utxos...) //newly created outputs and spent utxos
+	err = s.repo.SaveTX(ctx, &txRecord, upsertOutputs, newDataRecords)
```

Comment:
why we don't assume relationship between transaction and outputs and data.

1. I believe this should be responsiblity of GORM to map from struct relationship to database relationship.
2. if there would be more relationships this SaveTx method will have very long argument list.

Reply:
That's a good point.
I suggest to postpone this to the task when I try to link/connect those new `database.Models` to actual database with GORM.